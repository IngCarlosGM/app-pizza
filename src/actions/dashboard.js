import getData from "../helpers/getData";
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';

export const startLoadingStore = ( ) => {
    return ( dispatch ) => {
        dispatch(startLoading);
        
        getData().then(( resp ) => {
            dispatch(setStores( resp.data.response.stores ));

        }).catch( err => {
            dispatch( finishLoading );
        });

        dispatch(finishLoading);
    }
}

export const setStores = ( stores ) => ({
    type: types.dasboardStartLoadingStores,
    payload: {
        stores
    }
});

