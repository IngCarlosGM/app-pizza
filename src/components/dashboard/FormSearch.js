import React from 'react'
import PropTypes from 'prop-types'

export const FormSearch = ({ history, searchText, handleInputChange }) => {

    const handleSearch = (e) => {
        e.preventDefault();
        localStorage.setItem('lastPath', JSON.stringify(`?q=${ searchText }`) );
        history.push(`?q=${ searchText }`);
    }

    const handleKeyUp = (e) => {
        e.preventDefault();
        if( e.target.value === '') {
            history.push('/');
        }
    }

    return (
        <form onSubmit={ handleSearch }>
            <input 
                type="text"
                className="input-search input--block mt-third"
                placeholder="Buscar una tienda"
                name="searchText"
                value={ searchText }
                onChange={ handleInputChange }
                onKeyUp= { handleKeyUp }
            />

            <button
                id="btn_search"
                type="submit"
                className="btn__primary btn-block btn-responsive"
            >Buscar</button>
        </form>
    )
}

FormSearch.propTypes = {
    history: PropTypes.object.isRequired,
    searchText: PropTypes.string.isRequired,
    handleInputChange: PropTypes.func.isRequired
}