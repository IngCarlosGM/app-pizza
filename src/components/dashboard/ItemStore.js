import React from 'react';
import PropTypes from 'prop-types';

export const ItemStore = ({ itemStore }) => {
    return (
        <div className="container-item-store pointer">
            <div className="item-store">
                <img 
                    className="img-item-store"
                    src={ `../assets/store/store-${itemStore.id}.png` } 
                    alt="Logo pizza" 
                />
            </div>
            <div className="flex-column">
                <span className="title-item-store">{ itemStore.name }</span>
                <span className="addres-item-store">{ itemStore.address }</span>
            </div>
        </div>
    )
}

ItemStore.propTypes = {
    itemStore: PropTypes.object.isRequired
}