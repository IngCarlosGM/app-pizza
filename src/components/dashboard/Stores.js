import React from 'react';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLoadingStore } from '../../actions/dashboard';
import { ItemStore } from './ItemStore';

export const Stores = () => {

    const { stores }= useSelector(state => state.dashboard);
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch( startLoadingStore() );
    }, [dispatch]);

    return (
        <div className="grid-shop mt-secondary">
            {
                stores && stores.map( itemStore => {
                    return <ItemStore 
                                key={ `itemStore${ itemStore.id }` } 
                                itemStore={ itemStore }
                            />
                })
            }
        </div>
    )
}