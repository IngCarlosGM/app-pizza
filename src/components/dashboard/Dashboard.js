import React, { useEffect, useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import queryString from 'query-string';

import { NavBar } from '../nav-bar/NavBar';
import { RouteNavigation } from '../templates/RouteNavigation';
import { TitlePage } from '../templates/TitlePage';
import { DescriptionPage } from '../templates/DescriptionPage';
import { Stores } from './Stores';
import { Footer } from '../Footer';
import { useForm } from '../../hooks/useForm';
import { getStoresByName } from '../../selectors/getStoresByName';
import { FormSearch } from './FormSearch';
import { startLoadingStore } from '../../actions/dashboard';
import ModalCarrousel from '../modals/body-carrousel/ModalCarrousel';

export const Dashboard = ({ history }) => {

    const location = useLocation();
    const dispatch = useDispatch();

    const params = queryString.parse(location.search);
    const currentItem = parseInt(((params.id !== undefined) && params.id)  ||  0); //Store selected

    const { auth: user, dashboard: { stores } } = useSelector(state => state); //User authenticate and Store loaded
    const { q: query = '' } = queryString.parse( location.search ); //Text search on query
    const [ formValues, handleInputChange ] = useForm({
        searchText: query
    }); //Handle input to search stores
    const { searchText } = formValues;

    useEffect(() => {
        dispatch( startLoadingStore() );
        localStorage.setItem( 'user', JSON.stringify(user) );
        
        const logoAside = document.getElementById("logo-best-pizza-aside");
        if( logoAside && logoAside !== undefined ) {
            logoAside.style.display = "block";
        }
    }, [user, dispatch]);
    
    //Memorize results
    const storesFiltered = useMemo(() => getStoresByName( query, stores ), [query, stores]);

    return (
        <div className="scroll-y container-dashboard">
            {
                (currentItem !== 0) 
                    && <ModalCarrousel 
                        show={ params.show || 'false'}
                        currentItem={ currentItem }
                        stores={ storesFiltered }
                        history={ history }
                    />
            }
            <NavBar history={ history }/>
            <RouteNavigation route="Pizzerías"/>
            <TitlePage title="Tiendas" myStyle="mt-secondary"/>
            <DescriptionPage description="Escoge tu pizzería favorita"/>

            <FormSearch 
                history={ history }
                searchText={ searchText }
                handleInputChange={ handleInputChange }
            />

            {
                (storesFiltered.length === 0) 
                    && <div className="mt-third not-store-founds">No se han encontrado tiendas que coincidan con la búsqueda</div>
            }

            <Stores 
                stores={ storesFiltered }
                history={ history }
            />
            <Footer />
        </div>
    )
}
