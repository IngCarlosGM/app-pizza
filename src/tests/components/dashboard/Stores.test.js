import { mount } from "enzyme";
import { Router } from "react-router-dom";

import { Stores } from "../../../components/dashboard/Stores";
import dataApp from '../../data/dataApp.json';

describe('Testing Stores component', () => {

    const history = {
        location: {
            search: "url-test"
        },
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <Router history= { history }>
            <Stores 
                history= { history }
                stores= { dataApp.response.stores }
            />
        </Router>
    );
    
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have itemStores', () => {
        expect( wrapper.find('ItemStore').length ).toBe( dataApp.response.stores.length );
    });

});
