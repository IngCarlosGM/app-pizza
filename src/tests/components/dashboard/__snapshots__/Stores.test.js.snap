// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Stores component should render correctly 1`] = `
<div
  className="grid-shop mt-secondary"
>
  <a
    className="container-item-store pointer animate__bounceIn"
    onClick={[Function]}
  >
    <div
      className="item-store"
    >
      <img
        alt="Logo pizza"
        className="img-item-store"
        src="../assets/store/store-1.png"
      />
    </div>
    <div
      className="flex-column"
    >
      <span
        className="title-item-store"
      >
        Pizzeria Capitan America
      </span>
      <span
        className="addres-item-store"
      >
        Calle 1 #2-3
      </span>
    </div>
  </a>
  <a
    className="container-item-store pointer animate__bounceIn"
    onClick={[Function]}
  >
    <div
      className="item-store"
    >
      <img
        alt="Logo pizza"
        className="img-item-store"
        src="../assets/store/store-2.png"
      />
    </div>
    <div
      className="flex-column"
    >
      <span
        className="title-item-store"
      >
        Pizzeria Iron Man
      </span>
      <span
        className="addres-item-store"
      >
        Calle 2 #3-4
      </span>
    </div>
  </a>
  <a
    className="container-item-store pointer animate__bounceIn"
    onClick={[Function]}
  >
    <div
      className="item-store"
    >
      <img
        alt="Logo pizza"
        className="img-item-store"
        src="../assets/store/store-3.png"
      />
    </div>
    <div
      className="flex-column"
    >
      <span
        className="title-item-store"
      >
        Pizzeria Thor
      </span>
      <span
        className="addres-item-store"
      >
        Calle 3 #4-5
      </span>
    </div>
  </a>
  <a
    className="container-item-store pointer animate__bounceIn"
    onClick={[Function]}
  >
    <div
      className="item-store"
    >
      <img
        alt="Logo pizza"
        className="img-item-store"
        src="../assets/store/store-4.png"
      />
    </div>
    <div
      className="flex-column"
    >
      <span
        className="title-item-store"
      >
        Pizzeria Black Widow
      </span>
      <span
        className="addres-item-store"
      >
        Calle 4 #5-6
      </span>
    </div>
  </a>
  <a
    className="container-item-store pointer animate__bounceIn"
    onClick={[Function]}
  >
    <div
      className="item-store"
    >
      <img
        alt="Logo pizza"
        className="img-item-store"
        src="../assets/store/store-5.png"
      />
    </div>
    <div
      className="flex-column"
    >
      <span
        className="title-item-store"
      >
        Pizzeria Spider Man
      </span>
      <span
        className="addres-item-store"
      >
        Calle 5 #6-7
      </span>
    </div>
  </a>
</div>
`;
